resources/js/package.js

```diff
@@ -139,6 +139,24 @@ function init() {
 
                     return `/storage/${store}/resizes/${size}/magento${url}`
                 },
+
+                getCustomAttributeValue(object, code) {
+                    let attribute = object?.custom_attributes?.find((attribute) => attribute.code == code) ?? null
+
+                    if (!attribute) {
+                        return null
+                    }
+
+                    if ('value' in attribute) {
+                        return attribute.value
+                    }
+
+                    if ('selected_options' in attribute) {
+                        return Object.fromEntries(attribute.selected_options.map((option) => [option.value, option.label]))
```

Comment:
With the current implementation we would either get an array or direct value depending on the attribute.

How would this be used and would it be more preferable to return an array in both cases so the implementation of the function can always be the same?

Instead of having to check wether we received an array.

Reply:
I don't necessarily see this as a problem, because I figure when you're programming you would know what type of variable you're retrieving data from. You'd have the same issue when you're doing it the old way (except worse).

I definitely don't think it would be preferable to have this function always return one consistent type because the two possibilities are just fundamentally different data. If you wanted something for that, I think the best way would be to have two different functions.

Reply:
This is something we currently do for attribute_values on the cart

https://github.com/rapidez/core/blob/941cf91a332db3f87bf87f31a2238b701ce01f9d/resources/js/stores/useCart.js#L239

in order to make usage in template

https://github.com/rapidez/core/blob/e11744a4ed9105b367ca33dc38aa340f8518dbc2/resources/views/cart/item.blade.php#L36

more consistent

Reply:
I wouldn't say that's the same situation here though, as we've got an Object instead of an Array. However, something we could do is always return an object from `getCustomAttributes(data.customer)` and then you can use it like:

```js
getCustomAttributes(data.customer).my_attribute
getCustomAttributes(data.customer).my_options.option
```

maybe that's better? It kind ends up being the same thing, where `my_attribute` is a singular value but `my_options` is an Object.

We could also turn `my_attribute` here into an Object with a `value` so you always have to do `.value` after it... I'm not sure if that's any nicer.